spec-main/api-protocol-spec.ts

```diff
@@ -637,6 +638,22 @@ describe('protocol module', () => {
     });
   });
 
+  describe('protocol.registerSchemeAsPrivileged', () => {
+    it('does not crash on exit', async () => {
+      const appPath = path.join(__dirname, 'fixtures', 'api', 'custom-protocol-shutdown.js');
+      const appProcess = ChildProcess.spawn(process.execPath, ['--enable-logging', appPath]);
+      let output = '';
+      appProcess.stdout.on('data', data => { output += data; });
+      appProcess.stderr.on('data', data => { output += data; });
+      const [code] = await emittedOnce(appProcess, 'exit');
+      if (code !== 0) {
+        console.log(code, output);
+      }
+      expect(code).to.equal(0);
+      expect(output).to.not.contain('VALIDATION_ERROR_DESERIALIZATION_FAILED');
```

Comment:
```suggestion
      expect(stdout).to.not.contain('VALIDATION_ERROR_DESERIALIZATION_FAILED');
      expect(stderr).to.not.contain('VALIDATION_ERROR_DESERIALIZATION_FAILED');
```

stdout and stderr aren't necessarily synced, so this prevents potential mixing of output